src/main/java/net/minecraftforge/fmp/client/multipart/ICustomHighlightPart.java

```diff
@@ -0,0 +1,27 @@
+package net.minecraftforge.fmp.client.multipart;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraftforge.client.event.DrawBlockHighlightEvent;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.Multipart;
+
+/**
+ * Interface that adds custom hover highlight support to {@link IMultipart}. Hook to {@link DrawBlockHighlightEvent}
+ * adapted for multiparts.
+ *
+ * @see IMultipart
+ * @see Multipart
+ */
+public interface ICustomHighlightPart extends IMultipart
+{
+    
+    /**
+     * Draws this part's highlight.
+     */
+    @SideOnly(Side.CLIENT)
+    public boolean drawHighlight(RayTraceResult hit, EntityPlayer player, float partialTicks);
```

Comment:
I have nothing against it, but I wanted to keep as many clientside-only methods outside of IMultipart. I'll throw it in, though.